src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -441,9 +443,17 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     case Event(message: CallStartMessage, data) if !data.isPending(message.callKey) =>
       val updatedData = message.handleStatusPersist(this, data)
       val callKey = message.callKey
-
       fetchLocallyQualifiedInputs(callKey) match {
-        case Success(callInputs) => startActor(callKey, callInputs, message.startMode)
+        case Success(callInputs) =>
+          val updateDbCallInputs = globalDataAccess.updateCallInputs(workflow.id, callKey, callInputs)
+          updateDbCallInputs onSuccess {
+            case i =>
+              logger.debug(s"$i call input expression(s) updated in database.")
+              startActor(callKey, callInputs, message.startMode)
+          }
+          updateDbCallInputs onFailure {
```

Comment:
Also if it was an onComplete there's probably no need to assign it to updateDbCallInputs, just make the line and call onComplete I think?